File: nguyensao-javaspringboot/src/main/java/com/nguyensao/nguyensao_javaspringboot/service/PostService.java
package com.nguyensao.nguyensao_javaspringboot.service;

import java.util.List;

import com.nguyensao.nguyensao_javaspringboot.dto.PostDTO;
import com.nguyensao.nguyensao_javaspringboot.entity.Post;

public interface PostService {
    List<Post> index();

    List<Post> trash();

    Post show(Long id);

    Post store(PostDTO postDTO);

    Post update(Long id, PostDTO postDTO);

    void status(Long id);

    void delete(Long id);

    void restore(Long id);

    void destroy(Long id);

}

File: java/src/services/PostService.js
import httpAxios from "./httpAxios"

const PostService = {
    getList: async () => {
        return await httpAxios.get(`post`);
    },
    trash: async () => {
        return await httpAxios.get(`post/trash`);
    },
    show: async (id) => {
        return await httpAxios.get(`post/show/${id}`);
    },
    store: async (data) => {
        return await httpAxios.post(`post/store`, data);
    },
    update: async (data, id) => {
        return await httpAxios.post(`post/update/${id}`, data);
    },
    delete: async (id) => {
        return await httpAxios.get(`post/delete/${id}`);
    },
    restore: async (id) => {
        return await httpAxios.get(`post/restore/${id}`);
    },
    destroy: async (id) => {
        return await httpAxios.delete(`post/destroy/${id}`);
    },
}

export default PostService


